File: src/main/java/com/rkroom/blog/config/JWTToken.java
package com.rkroom.blog.config;

import org.apache.shiro.authc.AuthenticationToken;

public class JWTToken implements AuthenticationToken {
    // 密钥
    //不论执行的getPrincipal()还是getCredentials()都会返回jwt的token
    private String token;

    public JWTToken(String token) {
        this.token = token;
    }

    @Override
    public Object getPrincipal() {
        return token;
    }

    @Override
    public Object getCredentials() {
        return token;
    }

}

File: src/main/java/com/rkroom/blog/config/JWTToken.kt
package com.rkroom.blog.config

import org.apache.shiro.authc.AuthenticationToken

class JWTToken(
    // 密钥
    //不论执行的getPrincipal()还是getCredentials()都会返回jwt的token
    private val token: String?) : AuthenticationToken {

    override fun getPrincipal(): String? {
        return token
    }

    override fun getCredentials(): String? {
        return token
    }

}